snuba/datasets/transactions_processor.py

```diff
@@ -385,6 +392,12 @@ def _sanitize_contexts(
         transaction_ctx.pop("hash", None)
         transaction_ctx.pop("exclusive_time", None)
 
+        # The app_start_type is promoted as a column on a query level, no need to store it again
+        # in the context array.
+        app_ctx = sanitized_context.get("app", {})
+        if app_ctx is not None:
```

Comment:
This is the actual fix, most of the other code was already reviewed and approved in https://github.com/getsentry/snuba/pull/3127